flink-runtime/src/test/java/org/apache/flink/runtime/resourcemanager/active/ActiveResourceManagerFactoryTest.java

```diff
@@ -47,19 +46,19 @@
     private static final MemorySize TOTAL_FLINK_SIZE = MemorySize.ofMebiBytes(2 * 1024);
     private static final MemorySize TOTAL_PROCESS_SIZE = MemorySize.ofMebiBytes(3 * 1024);
 
-    private static Map<String, String> systemEnv;
+    private static Properties systemProperties;
 
     @BeforeClass
     public static void setupClass() {
-        systemEnv = System.getenv();
+        systemProperties = System.getProperties();
         System.clearProperty("flink.tests.disable-declarative");
         System.clearProperty("flink.tests.enable-fine-grained");
```

Comment:
Why do we have to clear these? Could we not just disable them in the configuration?

Reply:
In this test class, we expect getting different effective configurations, depending on whether fine-grained resource management is activated or not. Thus we want this class to have fully control which slot manager is used, without being affected by the properties for the CI stage.

Reply:
I get the intent, but if you'd set `config.set(ClusterOptions.ENABLE_DECLARATIVE_RESOURCE_MANAGEMENT, false);` to `testGetEffectiveConfigurationForResourceManagerCoarseGrained` then you are already in full control in regards to that behavior; the properties don't overrule explicit settings in the configuration.

As it is you could be affecting other tests.